common/student/activation.ts

```diff
@@ -0,0 +1,83 @@
+import {student as Student} from "@prisma/client";
+import {prisma} from "../prisma";
+import {dissolveMatch, dissolveProjectMatch} from "../match/dissolve";
+import {getTransactionLog} from "../transactionlog";
+import DeActivateEvent from "../transactionlog/types/DeActivateEvent";
+import {CourseState} from "../entity/Course";
+
+
+export async function deactivateStudent(student: Student) {
+    if (!student.active) {
+        throw new Error("Student was already deactivated");
+    }
+    // Dissolve matches for the student.
+    let matches = await prisma.match.findMany({
+        where: {
+            studentId: student.id,
+            dissolved: false
+        }
+    });
+
+    for (const match of matches) {
+        await dissolveMatch(match, 0, student);
+    }
+
+    let projectMatches = await prisma.project_match.findMany({
+        where: {
+            studentId: student.id,
+            dissolved: false
+        }
+    });
+
+    for (const match of projectMatches) {
+        await dissolveProjectMatch(match, 0, student);
+    }
+
+    //Delete course records for the student.
+    let studentCourses = await prisma.course_instructors_student.groupBy({
+        by: ['courseId'],
+        where: {
+            studentId: student.id
+        }
+    });
+
+    for (const course of studentCourses) {
+        await prisma.course_instructors_student.deleteMany({
+            where: {
+                studentId: student.id,
+                courseId: course.courseId
+            }
+        });
+        if (studentCourses.length === 1) {
+            await prisma.course.updateMany({
```

Comment:
Maybe this should be a `cancelCourse` function in `common/course`. As this is now the third place were we cancel courses with a different implmenentation (though your's is the best so far ;)) we might want to harmonize course cancellation at one place. Also if the student directly cancels the course (see https://github.com/corona-school/backend/blob/b59c8ee7476009eb0ad0bc7e6f1f04c28117279b/web/controllers/courseController/index.ts#L1975), we (a) send out emails and (b) insert `CancelSubcourseEvent` and `CancelCourseEvent` into the transaction log. I'm fine though if we leave this as is for the moment and harmonize this in a followup PR.